
<!DOCTYPE html>


<html lang="en" data-content_root="./" >

  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="viewport" content="width=device-width, initial-scale=1" />

    <title>Welcome to EPBD-BERT’s documentation! &#8212; EPBD-BERT 1.0.0 documentation</title>
  
  
  
  <script data-cfasync="false">
    document.documentElement.dataset.mode = localStorage.getItem("mode") || "";
    document.documentElement.dataset.theme = localStorage.getItem("theme") || "light";
  </script>
  
  <!-- Loaded before other Sphinx assets -->
  <link href="_static/styles/theme.css?digest=bd9e20870c6007c4c509" rel="stylesheet" />
<link href="_static/styles/bootstrap.css?digest=bd9e20870c6007c4c509" rel="stylesheet" />
<link href="_static/styles/pydata-sphinx-theme.css?digest=bd9e20870c6007c4c509" rel="stylesheet" />

  
  <link href="_static/vendor/fontawesome/6.5.1/css/all.min.css?digest=bd9e20870c6007c4c509" rel="stylesheet" />
  <link rel="preload" as="font" type="font/woff2" crossorigin href="_static/vendor/fontawesome/6.5.1/webfonts/fa-solid-900.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="_static/vendor/fontawesome/6.5.1/webfonts/fa-brands-400.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="_static/vendor/fontawesome/6.5.1/webfonts/fa-regular-400.woff2" />

    <link rel="stylesheet" type="text/css" href="_static/pygments.css?v=fa44fd50" />
    <link rel="stylesheet" type="text/css" href="_static/styles/sphinx-book-theme.css?v=384b581d" />
    <link rel="stylesheet" type="text/css" href="_static/graphviz.css?v=eafc0fe6" />
  
  <!-- Pre-loaded scripts that we'll load fully later -->
  <link rel="preload" as="script" href="_static/scripts/bootstrap.js?digest=bd9e20870c6007c4c509" />
<link rel="preload" as="script" href="_static/scripts/pydata-sphinx-theme.js?digest=bd9e20870c6007c4c509" />
  <script src="_static/vendor/fontawesome/6.5.1/js/all.min.js?digest=bd9e20870c6007c4c509"></script>

    <script src="_static/documentation_options.js?v=f9c1cefe"></script>
    <script src="_static/doctools.js?v=888ff710"></script>
    <script src="_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="_static/scripts/sphinx-book-theme.js?v=efea14e4"></script>
    <script>DOCUMENTATION_OPTIONS.pagename = 'index';</script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="epbd_bert" href="modules/modules.html" />
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta name="docsearch:language" content="en"/>
  </head>
  
  
  <body data-bs-spy="scroll" data-bs-target=".bd-toc-nav" data-offset="180" data-bs-root-margin="0px 0px -60%" data-default-mode="">

  
  
  <a id="pst-skip-link" class="skip-link" href="#main-content">Skip to main content</a>
  
  <div id="pst-scroll-pixel-helper"></div>

  
  <button type="button" class="btn rounded-pill" id="pst-back-to-top">
    <i class="fa-solid fa-arrow-up"></i>
    Back to top
  </button>

  
  <input type="checkbox"
          class="sidebar-toggle"
          name="__primary"
          id="__primary"/>
  <label class="overlay overlay-primary" for="__primary"></label>
  
  <input type="checkbox"
          class="sidebar-toggle"
          name="__secondary"
          id="__secondary"/>
  <label class="overlay overlay-secondary" for="__secondary"></label>
  
  <div class="search-button__wrapper">
    <div class="search-button__overlay"></div>
    <div class="search-button__search-container">
<form class="bd-search d-flex align-items-center"
      action="search.html"
      method="get">
  <i class="fa-solid fa-magnifying-glass"></i>
  <input type="search"
         class="form-control"
         name="q"
         id="search-input"
         placeholder="Search..."
         aria-label="Search..."
         autocomplete="off"
         autocorrect="off"
         autocapitalize="off"
         spellcheck="false"/>
  <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd>K</kbd></span>
</form></div>
  </div>

  <header>
  
    <div class="bd-header navbar navbar-expand-lg bd-navbar">
    </div>
  
  </header>

  <div class="bd-container">
    <div class="bd-container__inner bd-page-width">
      
      
      
        
      
      <div class="bd-sidebar-primary bd-sidebar">
        

  
  <div class="sidebar-header-items sidebar-primary__section">
    
    
    
    
  </div>
  
    <div class="sidebar-primary-items__start sidebar-primary__section">
        <div class="sidebar-primary-item">

  

<a class="navbar-brand logo" href="#">
  
  
  
  
  
  
    <p class="title logo__title">EPBD-BERT 1.0.0 documentation</p>
  
</a></div>
        <div class="sidebar-primary-item">

 <script>
 document.write(`
   <button class="btn navbar-btn search-button-field search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass"></i>
    <span class="search-button__default-text">Search</span>
    <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd class="kbd-shortcut__modifier">K</kbd></span>
   </button>
 `);
 </script></div>
        <div class="sidebar-primary-item"><nav class="bd-links bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item navbar-nav active">
        <ul class="nav bd-sidenav">
<li class="toctree-l1 has-children"><a class="reference internal" href="modules/modules.html">epbd_bert</a><input class="toctree-checkbox" id="toctree-checkbox-1" name="toctree-checkbox-1" type="checkbox"/><label class="toctree-toggle" for="toctree-checkbox-1"><i class="fa-solid fa-chevron-down"></i></label><ul>
<li class="toctree-l2 has-children"><a class="reference internal" href="modules/epbd_bert.html">epbd_bert package</a><input class="toctree-checkbox" id="toctree-checkbox-2" name="toctree-checkbox-2" type="checkbox"/><label class="toctree-toggle" for="toctree-checkbox-2"><i class="fa-solid fa-chevron-down"></i></label><ul>
<li class="toctree-l3"><a class="reference internal" href="modules/epbd_bert.datasets.html">epbd_bert.datasets package</a></li>
<li class="toctree-l3"><a class="reference internal" href="modules/epbd_bert.dnabert2_classifier.html">epbd_bert.dnabert2_classifier package</a></li>
<li class="toctree-l3"><a class="reference internal" href="modules/epbd_bert.dnabert2_epbd.html">epbd_bert.dnabert2_epbd package</a></li>
<li class="toctree-l3"><a class="reference internal" href="modules/epbd_bert.dnabert2_epbd_crossattn.html">epbd_bert.dnabert2_epbd_crossattn package</a></li>
<li class="toctree-l3"><a class="reference internal" href="modules/epbd_bert.utility.html">epbd_bert.utility package</a></li>
</ul>
</li>
</ul>
</li>
</ul>

    </div>
</nav></div>
    </div>
  
  
  <div class="sidebar-primary-items__end sidebar-primary__section">
  </div>
  
  <div id="rtd-footer-container"></div>


      </div>
      
      <main id="main-content" class="bd-main">
        
        

<div class="sbt-scroll-pixel-helper"></div>

          <div class="bd-content">
            <div class="bd-article-container">
              
              <div class="bd-header-article">
<div class="header-article-items header-article__inner">
  
    <div class="header-article-items__start">
      
        <div class="header-article-item"><label class="sidebar-toggle primary-toggle btn btn-sm" for="__primary" title="Toggle primary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
  <span class="fa-solid fa-bars"></span>
</label></div>
      
    </div>
  
  
    <div class="header-article-items__end">
      
        <div class="header-article-item">

<div class="article-header-buttons">





<div class="dropdown dropdown-download-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Download this page">
    <i class="fas fa-download"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="_sources/index.rst" target="_blank"
   class="btn btn-sm btn-download-source-button dropdown-item"
   title="Download source file"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file"></i>
  </span>
<span class="btn__text-container">.rst</span>
</a>
</li>
      
      
      
      
      <li>
<button onclick="window.print()"
  class="btn btn-sm btn-download-pdf-button dropdown-item"
  title="Print to PDF"
  data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file-pdf"></i>
  </span>
<span class="btn__text-container">.pdf</span>
</button>
</li>
      
  </ul>
</div>




<button onclick="toggleFullScreen()"
  class="btn btn-sm btn-fullscreen-button"
  title="Fullscreen mode"
  data-bs-placement="bottom" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-expand"></i>
  </span>

</button>



<script>
document.write(`
  <button class="btn btn-sm navbar-btn theme-switch-button" title="light/dark" aria-label="light/dark" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <span class="theme-switch nav-link" data-mode="light"><i class="fa-solid fa-sun fa-lg"></i></span>
    <span class="theme-switch nav-link" data-mode="dark"><i class="fa-solid fa-moon fa-lg"></i></span>
    <span class="theme-switch nav-link" data-mode="auto"><i class="fa-solid fa-circle-half-stroke fa-lg"></i></span>
  </button>
`);
</script>


<script>
document.write(`
  <button class="btn btn-sm navbar-btn search-button search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass fa-lg"></i>
  </button>
`);
</script>
<label class="sidebar-toggle secondary-toggle btn btn-sm" for="__secondary"title="Toggle secondary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <span class="fa-solid fa-list"></span>
</label>
</div></div>
      
    </div>
  
</div>
</div>
              
              

<div id="jb-print-docs-body" class="onlyprint">
    <h1>Welcome to EPBD-BERT’s documentation!</h1>
    <!-- Table of contents -->
    <div id="print-main-content">
        <div id="jb-print-toc">
            
            <div>
                <h2> Contents </h2>
            </div>
            <nav aria-label="Page">
                <ul class="visible nav section-nav flex-column">
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#">Welcome to EPBD-BERT’s documentation!</a></li>
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#resources">Resources</a></li>
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#installation">Installation</a></li>
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#data-preprocessing-steps">Data Preprocessing Steps</a></li>
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#preprocessed-dataset-loading">Preprocessed dataset loading</a></li>
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#training-and-testing-the-developed-models">Training and testing the developed models</a></li>
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#acknowledgements">Acknowledgements</a></li>
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#copyright-notice">Copyright notice</a></li>
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#license">License</a></li>
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#id1">Documentation</a></li>
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#indices-and-tables">Indices and tables</a></li>
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#authors">Authors</a></li>
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#how-to-cite-epbd-bert">How to cite EPBD-BERT?</a></li>
</ul>

            </nav>
        </div>
    </div>
</div>

              
                
<div id="searchbox"></div>
                <article class="bd-article">
                  
  <section id="welcome-to-epbd-bert-s-documentation">
<h1>Welcome to EPBD-BERT’s documentation!<a class="headerlink" href="#welcome-to-epbd-bert-s-documentation" title="Link to this heading">#</a></h1>
<p>This repository corresponds to the article titled <strong>“Advancing Transcription Factor Binding Site Prediction Using DNA Breathing Dynamics and Sequence Transformers via Cross Attention”</strong>.</p>
<figure class="align-center" id="id2">
<a class="reference internal image-reference" href="_images/EPBD_Arch.jpg"><img alt="_images/EPBD_Arch.jpg" src="_images/EPBD_Arch.jpg" style="width: 90%;" /></a>
<figcaption>
<p><span class="caption-text">Figure 1: Overview of the proposed EPBDxBERT framework.</span><a class="headerlink" href="#id2" title="Link to this image">#</a></p>
</figcaption>
</figure>
<p>Understanding the impact of genomic variants on transcription factor binding and gene regulation remains a key area of
research, with implications for unraveling the complex mechanisms underlying various functional effects. This software
framework delves into the role of DNA’s biophysical properties, including thermodynamic stability, shape, and flexibility in
transcription factor (TF) binding. In this library, we have developed a multi-modal deep learning model integrating these
properties with DNA sequence data. Trained on ChIP-Seq (chromatin immunoprecipitation sequencing) data in-vivo
involving 690 TF-DNA binding events in human genome, our model significantly improves prediction performance in over
660 binding events, with up to 9.6% increase in AUROC metric compared to the baseline model when using no DNA
biophysical properties explicitly. Further, we expanded our analysis to in-vitro high-throughput Systematic Evolution
of Ligands by Exponential enrichment (SELEX) dataset, comparing our model with
established frameworks. The inclusion of EPBD features consistently improved TF binding predictions across different cell
lines in these datasets. Notably, for complex ChIP-Seq datasets, integrating DNABERT2 with a cross-attention mechanism
provided greater predictive capabilities and insights into the mechanisms of disease-related non-coding variants found in
genome-wide association studies. This work highlights the importance of DNA biophysical characteristics in TF binding
and the effectiveness of multi-modal deep learning models in gene regulation studies</p>
</section>
<section id="resources">
<h1>Resources<a class="headerlink" href="#resources" title="Link to this heading">#</a></h1>
<ul class="simple">
<li><p><a class="reference external" href="https://www.biorxiv.org/content/10.1101/2024.01.16.575935v2.abstract">Paper</a></p></li>
<li><p><a class="reference external" href="https://github.com/lanl/EPBD-BERT">Code</a></p></li>
<li><p><a class="reference external" href="https://lanl.github.io/EPBD-BERT/">Documentation</a></p></li>
<li><p><a class="reference external" href="https://github.com/lanl/EPBD-BERT/tree/main/analysis">Analysis Notebooks</a></p></li>
</ul>
</section>
<section id="installation">
<h1>Installation<a class="headerlink" href="#installation" title="Link to this heading">#</a></h1>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span><span class="c1"># Installation of virtural environment</span>
git<span class="w"> </span>clone<span class="w"> </span>https://github.com/lanl/EPBD-BERT.git
<span class="nb">cd</span><span class="w"> </span>EPBD-BERT
conda<span class="w"> </span>create<span class="w"> </span>-c<span class="w"> </span>conda-forge<span class="w"> </span>-p<span class="w"> </span>.venvs/epbd_bert_condavenv_test1<span class="w"> </span><span class="nv">python</span><span class="o">=</span><span class="m">3</span>.11<span class="w"> </span>-y
conda<span class="w"> </span>activate<span class="w"> </span>.venvs/epbd_bert_condavenv_test1
python<span class="w"> </span>setup.py<span class="w"> </span>install

conda<span class="w"> </span>install<span class="w"> </span>-c<span class="w"> </span>conda-forge<span class="w"> </span>scikit-learn<span class="w"> </span>scipy<span class="w"> </span>-y
pip<span class="w"> </span>uninstall<span class="w"> </span>triton<span class="w"> </span><span class="c1"># We did not utilize triton for underlying hardware dependency</span>

<span class="c1"># To deactivate and remove the venv</span>
conda<span class="w"> </span>deactivate
conda<span class="w"> </span>remove<span class="w"> </span>--name<span class="w"> </span>epbd_bert_condavenv_test1<span class="w"> </span>--all<span class="w"> </span>-y
conda<span class="w"> </span>remove<span class="w"> </span>-p<span class="w"> </span>.venvs/epbd_bert_condavenv_test1<span class="w"> </span>--all<span class="w"> </span>-y
</pre></div>
</div>
</section>
<section id="data-preprocessing-steps">
<h1>Data Preprocessing Steps<a class="headerlink" href="#data-preprocessing-steps" title="Link to this heading">#</a></h1>
<p>The ‘data_preprocessing’ directory holds all the data generation steps and divided into modules for data generation and bug tracking. We utilized ‘[bedtools](<a class="reference external" href="https://bedtools.readthedocs.io/en/latest/">https://bedtools.readthedocs.io/en/latest/</a>)’ software for genome operation. Follow the [bedtools installation guide](<a class="reference external" href="https://bedtools.readthedocs.io/en/latest/content/installation.html">https://bedtools.readthedocs.io/en/latest/content/installation.html</a>). We also provide a bare minimum script that downloads the pre-compiled binary of the software into the <em>bedtools</em> directory:</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>bash<span class="w"> </span>setup_bedtools.sh
<span class="nb">export</span><span class="w"> </span><span class="nv">PATH</span><span class="o">=</span><span class="nv">$PATH</span>:<span class="k">$(</span><span class="nb">pwd</span><span class="k">)</span>/bedtools
</pre></div>
</div>
<table class="table">
<colgroup>
<col style="width: 50.0%" />
<col style="width: 50.0%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p>Step</p></th>
<th class="head"><p>Scripts</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>Download human genome assembly (GRCh37/hg19) and <a class="reference external" href="https://genome.ucsc.edu/cgi-bin/hgTrackUi?hgsid=2215774794_SHfvFO0XVRMcn6xaqOTugAa1Faf1&amp;c=chr1&amp;g=wgEncodeAwgTfbsUniform">uniform TFBS</a></p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">0_download_data.py</span></code></p></td>
</tr>
<tr class="row-odd"><td><p>Preprocess TFBS narrowpeak files and human genome</p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">1_preprocess_narrowPeaks_and_humanGenome.sh</span></code></p></td>
</tr>
<tr class="row-even"><td><p>Overlapping computation for label association</p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">2.1_compute_overlappings_job.sh</span></code> <code class="docutils literal notranslate"><span class="pre">2.2_compute_overlappings.sh</span></code> <code class="docutils literal notranslate"><span class="pre">3_postprocess.sh</span></code></p></td>
</tr>
<tr class="row-odd"><td></td>
<td></td>
</tr>
<tr class="row-even"><td></td>
<td></td>
</tr>
<tr class="row-odd"><td></td>
<td></td>
</tr>
<tr class="row-even"><td></td>
<td></td>
</tr>
</tbody>
</table>
</section>
<section id="preprocessed-dataset-loading">
<h1>Preprocessed dataset loading<a class="headerlink" href="#preprocessed-dataset-loading" title="Link to this heading">#</a></h1>
<p>Preprocessed dataset can be downloaded from here (will be provided).</p>
<div class="line-block">
<div class="line">Dataset Module  | Usage |</div>
<div class="line">:— | :— |</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">`epbd_bert.datasets.sequence_dataset`</span></code>  | Loads sequence only dataset  |</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">`epbd_bert.datasets.sequence_epbd_dataset`</span></code> | Loads sequence and EPBD (flat) features |</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">`epbd_bert.datasets.sequence_epbd_multimodal_dataset`</span></code> | Loads sequence and EPBD (matrix) features |</div>
</div>
<p>Note: There are some other dataset modules. Each module provides example running instructions at the bottom.</p>
</section>
<section id="training-and-testing-the-developed-models">
<h1>Training and testing the developed models<a class="headerlink" href="#training-and-testing-the-developed-models" title="Link to this heading">#</a></h1>
<div class="line-block">
<div class="line">Model Module | Usage |</div>
<div class="line">:— | :— |</div>
<div class="line">DNABERT2-finetuned | |</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">`epbd_bert.dnabert2_classifier.train_lightning`</span></code> | Train DNABERT2 using train/validation split |</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">`epbd_bert.dnabert2_classifier.test`</span></code> | Test finetuned DNABERT2 on test split |</div>
<div class="line">VanillaEPBD-DNABERT2-coordflip | |</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">`epbd_bert.dnabert2_epbd.train_lightning`</span></code> | Train VanillaEPBD-DNABERT2 using train/validation split |</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">`epbd_bert.dnabert2_epbd.test`</span></code> | Test VanillaEPBD-DNABERT2 on test split |</div>
<div class="line">EPBDxDNABERT-2 | |</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">`epbd_bert.dnabert2_epbd_crossattn.train_lightning`</span></code> | Train EPBDxDNABERT-2 using train/validation split |</div>
<div class="line"><code class="docutils literal notranslate"><span class="pre">`epbd_bert.dnabert2_epbd_crossattn.test`</span></code> | Test EPBDxDNABERT-2 on test split  |</div>
</div>
<p>Note: Details of each model with other ablation study can be found in the [Paper](<a class="reference external" href="https://www.biorxiv.org/content/10.1101/2024.01.16.575935v2.abstract">https://www.biorxiv.org/content/10.1101/2024.01.16.575935v2.abstract</a>). To run train/test: <code class="docutils literal notranslate"><span class="pre">`python</span> <span class="pre">-m</span> <span class="pre">epbd_bert.dnabert2_classifier.test`</span></code>.</p>
</section>
<section id="acknowledgements">
<h1>Acknowledgements<a class="headerlink" href="#acknowledgements" title="Link to this heading">#</a></h1>
<p>Los Alamos National Lab (LANL), T-1</p>
</section>
<section id="copyright-notice">
<h1>Copyright notice<a class="headerlink" href="#copyright-notice" title="Link to this heading">#</a></h1>
<p>© 2024. Triad National Security, LLC. All rights reserved.
This program was produced under U.S. Government contract 89233218CNA000001 for Los Alamos National Laboratory (LANL), which is operated by Triad National Security, LLC for the U.S. Department of Energy/National Nuclear Security Administration. All rights in the program are reserved by Triad National Security, LLC, and the U.S. Department of Energy/National Nuclear Security Administration. The Government is granted for itself and others acting on its behalf a nonexclusive, paid-up, irrevocable worldwide license in this material to reproduce, prepare. derivative works, distribute copies to the public, perform publicly and display publicly, and to permit others to do so.
LANL O#4717</p>
</section>
<section id="license">
<h1>License<a class="headerlink" href="#license" title="Link to this heading">#</a></h1>
<p>This program is Open-Source under the BSD-3 License.</p>
<ol class="arabic simple">
<li><p>Redistribution and use in source and binary forms, with or without modification, are permitted provided that the following conditions are met:</p></li>
<li><p>Redistributions of source code must retain the above copyright notice, this list of conditions and the following disclaimer.</p></li>
<li><p>Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the following disclaimer in the documentation and/or other materials provided with the distribution.</p></li>
</ol>
<p>Neither the name of the copyright holder nor the names of its contributors may be used to endorse or promote products derived from this software without specific prior written permission.</p>
<p>THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS “AS IS” AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.</p>
</section>
<section id="id1">
<h1>Documentation<a class="headerlink" href="#id1" title="Link to this heading">#</a></h1>
<div class="toctree-wrapper compound">
<ul>
<li class="toctree-l1"><a class="reference internal" href="modules/modules.html">epbd_bert</a><ul>
<li class="toctree-l2"><a class="reference internal" href="modules/epbd_bert.html">epbd_bert package</a><ul>
<li class="toctree-l3"><a class="reference internal" href="modules/epbd_bert.html#subpackages">Subpackages</a><ul>
<li class="toctree-l4"><a class="reference internal" href="modules/epbd_bert.datasets.html">epbd_bert.datasets package</a></li>
<li class="toctree-l4"><a class="reference internal" href="modules/epbd_bert.dnabert2_classifier.html">epbd_bert.dnabert2_classifier package</a></li>
<li class="toctree-l4"><a class="reference internal" href="modules/epbd_bert.dnabert2_epbd.html">epbd_bert.dnabert2_epbd package</a></li>
<li class="toctree-l4"><a class="reference internal" href="modules/epbd_bert.dnabert2_epbd_crossattn.html">epbd_bert.dnabert2_epbd_crossattn package</a></li>
<li class="toctree-l4"><a class="reference internal" href="modules/epbd_bert.utility.html">epbd_bert.utility package</a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="modules/epbd_bert.html#submodules">Submodules</a></li>
<li class="toctree-l3"><a class="reference internal" href="modules/epbd_bert.html#module-epbd_bert.models_factory">epbd_bert.models_factory module</a><ul>
<li class="toctree-l4"><a class="reference internal" href="modules/epbd_bert.html#epbd_bert.models_factory.get_model_and_dataloader"><code class="docutils literal notranslate"><span class="pre">get_model_and_dataloader()</span></code></a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="modules/epbd_bert.html#module-epbd_bert.path_configs">epbd_bert.path_configs module</a></li>
<li class="toctree-l3"><a class="reference internal" href="modules/epbd_bert.html#module-epbd_bert.version">epbd_bert.version module</a></li>
<li class="toctree-l3"><a class="reference internal" href="modules/epbd_bert.html#module-epbd_bert">Module contents</a></li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
</section>
<section id="indices-and-tables">
<h1>Indices and tables<a class="headerlink" href="#indices-and-tables" title="Link to this heading">#</a></h1>
<ul class="simple">
<li><p><a class="reference internal" href="genindex.html"><span class="std std-ref">Index</span></a></p></li>
<li><p><a class="reference internal" href="py-modindex.html"><span class="std std-ref">Module Index</span></a></p></li>
<li><p><a class="reference internal" href="search.html"><span class="std std-ref">Search Page</span></a></p></li>
</ul>
</section>
<section id="authors">
<h1>Authors<a class="headerlink" href="#authors" title="Link to this heading">#</a></h1>
<ul class="simple">
<li><p><a class="reference external" href="mailto:akabir4&#37;&#52;&#48;gmu&#46;edu">Anowarul Kabir</a>: Computer Sciece, George Mason University</p></li>
<li><p><a class="reference external" href="mailto:ceodspspectrum&#37;&#52;&#48;lanl&#46;gov">Manish Bhattarai</a>: Theoretical Division, Los Alamos National Laboratory</p></li>
<li><p><a class="reference external" href="mailto:kor&#37;&#52;&#48;lanl&#46;gov">Kim Rasmussen</a>: Theoretical Division, Los Alamos National Laboratory</p></li>
<li><p><a class="reference external" href="mailto:ashehu&#37;&#52;&#48;gmu&#46;edu">Amarda Shehu</a>: Computer Sciece, George Mason University</p></li>
<li><p><a class="reference external" href="mailto:arb&#37;&#52;&#48;lanl&#46;gov">Alan Bishop</a>: Theoretical Division, Los Alamos National Laboratory</p></li>
<li><p><a class="reference external" href="mailto:boian&#37;&#52;&#48;lanl&#46;gov">Boian S<span>&#46;</span> Alexandrov</a>: Theoretical Division, Los Alamos National Laboratory</p></li>
<li><p><a class="reference external" href="mailto:Anny_Usheva&#37;&#52;&#48;brown&#46;edu">Anny Usheva</a>: Surgery, Rhode Island Hospital and Brown University</p></li>
</ul>
</section>
<section id="how-to-cite-epbd-bert">
<h1>How to cite EPBD-BERT?<a class="headerlink" href="#how-to-cite-epbd-bert" title="Link to this heading">#</a></h1>
<div class="highlight-console notranslate"><div class="highlight"><pre><span></span><span class="go">@article{kabir2024advancing,</span>
<span class="go">   title    = {Advancing Transcription Factor Binding Site Prediction Using DNA Breathing Dynamics and Sequence Transformers via Cross Attention},</span>
<span class="go">   author   = {Kabir, Anowarul and</span>
<span class="go">               Bhattarai, Manish and</span>
<span class="go">               Rasmussen, Kim {\O} and</span>
<span class="go">               Shehu, Amarda and</span>
<span class="go">               Bishop, Alan R and</span>
<span class="go">               Alexandrov, Boian and</span>
<span class="go">               Usheva, Anny},</span>
<span class="go">   journal  = {bioRxiv},</span>
<span class="go">   pages    = {2024--01},</span>
<span class="go">   year     = {2024},</span>
<span class="go">   publisher= {Cold Spring Harbor Laboratory},</span>
<span class="go">   doi      = {10.5281/zenodo.11130474},</span>
<span class="go">   url      = {https://www.biorxiv.org/content/10.1101/2024.01.16.575935v2}</span>
<span class="go">   }</span>
</pre></div>
</div>
</section>


                </article>
              

              
              
              
              
                <footer class="prev-next-footer">
                  
<div class="prev-next-area">
    <a class="right-next"
       href="modules/modules.html"
       title="next page">
      <div class="prev-next-info">
        <p class="prev-next-subtitle">next</p>
        <p class="prev-next-title">epbd_bert</p>
      </div>
      <i class="fa-solid fa-angle-right"></i>
    </a>
</div>
                </footer>
              
            </div>
            
            
              
                <div class="bd-sidebar-secondary bd-toc"><div class="sidebar-secondary-items sidebar-secondary__inner">


  <div class="sidebar-secondary-item">
  <div class="page-toc tocsection onthispage">
    <i class="fa-solid fa-list"></i> Contents
  </div>
  <nav class="bd-toc-nav page-toc">
    <ul class="visible nav section-nav flex-column">
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#">Welcome to EPBD-BERT’s documentation!</a></li>
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#resources">Resources</a></li>
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#installation">Installation</a></li>
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#data-preprocessing-steps">Data Preprocessing Steps</a></li>
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#preprocessed-dataset-loading">Preprocessed dataset loading</a></li>
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#training-and-testing-the-developed-models">Training and testing the developed models</a></li>
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#acknowledgements">Acknowledgements</a></li>
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#copyright-notice">Copyright notice</a></li>
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#license">License</a></li>
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#id1">Documentation</a></li>
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#indices-and-tables">Indices and tables</a></li>
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#authors">Authors</a></li>
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#how-to-cite-epbd-bert">How to cite EPBD-BERT?</a></li>
</ul>

  </nav></div>

</div></div>
              
            
          </div>
          <footer class="bd-footer-content">
            
<div class="bd-footer-content__inner container">
  
  <div class="footer-item">
    
<p class="component-author">
By Anowarul Kabir, Manish Bhattarai, Kim Rasmussen, Amarda Shehu, Alan Bishop, Boian Alexandrov and Anny Usheva
</p>

  </div>
  
  <div class="footer-item">
    

  <p class="copyright">
    
      © Copyright 2023, LANL.
      <br/>
    
  </p>

  </div>
  
  <div class="footer-item">
    
  </div>
  
  <div class="footer-item">
    
  </div>
  
</div>
          </footer>
        

      </main>
    </div>
  </div>
  
  <!-- Scripts loaded after <body> so the DOM is not blocked -->
  <script src="_static/scripts/bootstrap.js?digest=bd9e20870c6007c4c509"></script>
<script src="_static/scripts/pydata-sphinx-theme.js?digest=bd9e20870c6007c4c509"></script>

  <footer class="bd-footer">
  </footer>
  </body>
</html>